packages/devtools_app/lib/src/charts/chart.dart

```diff
@@ -0,0 +1,588 @@
+    displayTime = false,
+  }) {
+    final xAxisYCoord =
+        chartController.yCanvasChart + chartController.canvasChartHeight;
+
+    // Draw vertical tick (short or long).
+    canvas.drawLine(
+      Offset(xTickCoord, xAxisYCoord),
+      Offset(xTickCoord, xAxisYCoord + (shortTick ? 2 : 6)),
+      axis,
+    );
+
+    if (displayTime) {
+      final tp = createText(prettyTimestamp(timestamp), 1);
+      tp.paint(
+        canvas,
+        Offset(xTickCoord - 30, xAxisYCoord + 15 - tp.height / 2),
+      );
+    }
+  }
+
+  void drawText(String key, Canvas canvas, double x, double y) {
+    final tp = createText(key, 1);
+    tp.paint(canvas, Offset(x, y - tp.height / 2));
+  }
+
+  TextPainter createText(String key, double scale) {
+    final span = TextSpan(
+      style: TextStyle(color: Colors.grey[600]),
+      text: key,
+    );
+    final tp = TextPainter(
+      text: span,
+      textAlign: TextAlign.left,
+      textScaleFactor: scale,
+      textDirection: TextDirection.ltr,
+    );
+    tp.layout();
+    return tp;
+  }
+
+  void drawCircle(
+    Canvas canvas,
+    PaintCharacteristics characteristics,
+    double x,
+    double y,
+  ) {
+    final paint = Paint()
+      ..style = characteristics.symbol == ChartSymbol.disc
+          ? PaintingStyle.fill
+          : PaintingStyle.stroke
+      ..strokeWidth = characteristics.strokeWidth
+      ..color = characteristics.color;
+
+    canvas.drawCircle(Offset(x, y), characteristics.diameter, paint);
+  }
+
+  // TODO(terry): Use bezier path.
+  void drawDashed(Canvas canvas, PaintCharacteristics characteristics, double x,
+      double y, double tickWidth) {
+    assert(characteristics.symbol == ChartSymbol.dashedLine);
+    drawLine(
+      canvas,
+      characteristics,
+      x,
+      y,
+      tickWidth,
+    );
+  }
+
+  void drawConnectedLine(
+    Canvas canvas,
+    PaintCharacteristics characteristics,
+    double startX,
+    double startY,
+    double endX,
+    double endY,
+  ) {
+    final paint = Paint()
+      ..style = PaintingStyle.stroke
+      ..strokeWidth = characteristics.strokeWidth
+      ..color = characteristics.color;
+
+    canvas.drawLine(Offset(startX, startY), Offset(endX, endY), paint);
+  }
+
+  void drawLine(
+    Canvas canvas,
+    PaintCharacteristics characteristics,
+    double x,
+    double y,
+    double tickWidth,
+  ) {
+    final paint = Paint()
+      ..style = PaintingStyle.stroke
+      ..strokeWidth = characteristics.strokeWidth
+      ..color = characteristics.color;
+
+    canvas.drawLine(Offset(x, y), Offset(x + tickWidth, y), paint);
+  }
+
+  void drawFillArea(
+    Canvas canvas,
+    PaintCharacteristics characteristics,
+    double x0,
+    double y0,
+    double y0Bottom,
+    double x1,
+    double y1,
+    double y1Bottom,
+  ) {
+    final paint = Paint()
+      ..style = PaintingStyle.fill
+      ..strokeWidth = characteristics.strokeWidth
+      ..color = characteristics.color.withAlpha(100);
```

Comment:
withAlpha(100) isn't needed.

Reply:
I plan to add a light grid below (paper of the chart) that show's movement/point of reference with relation to the x-axis moving (probably only a few vertical lines).